vspace/src/main/java/edu/asu/diging/vspace/web/staff/EditReferenceController.java

```diff
@@ -0,0 +1,69 @@
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.servlet.mvc.support.RedirectAttributes;
+
+import edu.asu.diging.vspace.core.model.IReference;
+import edu.asu.diging.vspace.core.model.impl.Reference;
+import edu.asu.diging.vspace.core.services.IReferenceManager;
+
+@Controller
+public class EditReferenceController {
+    
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+
+    @Autowired
+    private IReferenceManager referenceManager;
+    
+    @RequestMapping(value = "/staff/reference/{referenceId}/edit", method = RequestMethod.GET)
+    public String show(Model model, @PathVariable("referenceId") String referenceId, RedirectAttributes attributes) {
+        IReference reference = referenceManager.getReferenceById(referenceId);
+        model.addAttribute("referenceData", reference);
+        model.addAttribute("referenceId", referenceId);
+        return "staff/references/edit";
+    }
+    
+    @RequestMapping(value = "/staff/reference/{referenceId}/edit", method = RequestMethod.POST)
+    public String save(@ModelAttribute Reference refData, @PathVariable("referenceId") String referenceId, 
+            RedirectAttributes attributes) {
+        IReference reference = referenceManager.getReferenceById(referenceId);
+        refData.setId(reference.getId());
+        referenceManager.updateReference(refData);
+        return "redirect:/staff/display/reference/{referenceId}";
+    }
+
+    @RequestMapping(value = "/staff/module/{moduleId}/slide/{id}/biblio/{biblioId}/reference/edit", method = RequestMethod.POST)
+    public ResponseEntity<String> editReference(@PathVariable("id") String slideId,
+            @PathVariable("moduleId") String moduleId, @PathVariable("biblioId") String biblioId, 
+            @RequestBody Reference ref) throws IOException {
+        IReference reference = referenceManager.getReference(ref.getId());
+        reference.setTitle(ref.getTitle());
+        reference.setAuthor(ref.getAuthor());
+        reference.setYear(reference.getYear());
+        reference.setJournal(ref.getJournal());
+        reference.setUrl(ref.getUrl());
+        reference.setVolume(ref.getVolume());
+        reference.setIssue(ref.getIssue());
+        reference.setPages(ref.getPages());
+        reference.setEditors(ref.getEditors());
+        reference.setType(ref.getType());
+        reference.setNote(ref.getNote());
+        referenceManager.updateReference((Reference) reference);
```

Comment:
Don't need this method anymore. Deleting it.

Reply:
Need this method to edit references on slide page.

Reply:
but why is this not doing the same thing as the other method since its goal is the same, isn't it? Why does this separately set all the properties and not saving the reference object like the other?